rust/src/transformations/polars/make_select_trans/mod.rs

```diff
@@ -0,0 +1,273 @@
+use polars::prelude::*;
+
+use crate::core::{Function, StabilityMap, Transformation};
+use crate::domains::{Context, ExprDomain, LazyFrameDomain};
+use crate::error::*;
+use crate::metrics::{IntDistance, SymmetricDistance};
+
+/// Make a Transformation that applies list of transformations to a Lazy Frame.
+///
+pub fn make_select_trans(
+    input_domain: LazyFrameDomain,
+    input_metric: SymmetricDistance,
+    transformations: Vec<
+        Transformation<ExprDomain, ExprDomain, SymmetricDistance, SymmetricDistance>,
+    >,
+) -> Fallible<Transformation<LazyFrameDomain, LazyFrameDomain, SymmetricDistance, SymmetricDistance>> where
+{
+    let fist_transformation = transformations
+        .first()
+        .ok_or_else(|| err!(MakeTransformation, "transformation list cannot be empty"))?;
+
+    let output_domain = fist_transformation.output_domain.lazy_frame_domain.clone();
+
+    let output_metric = fist_transformation.output_metric.clone();
+
+    if transformations
+        .iter()
+        .any(|t| t.input_domain.lazy_frame_domain != input_domain)
+    {
+        return fallible!(MakeTransformation, "input domains do not match");
+    }
+
+    if transformations
+        .iter()
+        .any(|t| t.output_domain.context != Context::Select)
+    {
+        return fallible!(
+            MakeTransformation,
+            "transformation is not in select context"
+        );
+    }
+
+    if transformations
+        .iter()
+        .any(|t| t.output_domain.lazy_frame_domain != output_domain)
+    {
+        return fallible!(
+            MakeTransformation,
+            "transformations' output domains do not match"
+        );
+    }
```

Comment:
We are expecting each transformation's output domain to be different. The final output domain should be constructed out of the output domains of each of the transformations.

```rust
let output_domain = LazyFrameDomain::new(transformations.iter()
    .map(|t| &t.output_domain)
    .map(|d| d.lazy_frame_domain.try_column(d.selected_column()?))
    .collect::<Fallible<Vec<SeriesDomain>>>()?)
```